posthog/cache.py

```diff
@@ -6,32 +6,37 @@
 
 redis_instance = None  # type: Optional[redis.Redis]
 
-if settings.TEST:
-    redis_instance = fakeredis.FakeStrictRedis()
-elif settings.REDIS_URL:
-    redis_instance = redis.from_url(settings.REDIS_URL, db=0)
+
+def get_redis_instance() -> redis.Redis:
+    global redis_instance
+
+    if redis_instance:
+        return redis_instance
+
+    if settings.TEST:
+        redis_instance = fakeredis.FakeStrictRedis()
+    elif settings.REDIS_URL:
+        redis_instance = redis.from_url(settings.REDIS_URL, db=0)
+
+    if not redis_instance:
```

Comment:
Do we allow people to run posthog w/o redis currently?

Reply:
Nope, we `raise ImproperlyConfigured` in settings if there's no redis. This unreachable code is just so mypy would always return `redis.Redis` for this function and not `Optional[redis.Redis]`